v2/internal/reconcilers/azure_generic_arm_reconciler.go

```diff
@@ -93,7 +94,8 @@ func NewAzureDeploymentReconciler(
 	resourceResolver *genruntime.Resolver,
 	positiveConditions *conditions.PositiveConditionBuilder,
 	cfg config.Values,
-	rand *rand.Rand) *AzureDeploymentReconciler {
+	rand *rand.Rand,
+	extensions genruntime.ResourceExtension) *AzureDeploymentReconciler {
```

Comment:
Why does this have a plural name for a single instance of the interface?

Reply:
Ah, I was thinking that there was only one _instance_ being passed in, so it should have a singular name, regardless of how many extension-points it implements.